Start Viewer with --viewer

soneryaldiz opened the pull request.

stevenmburns commented on align/main.py:
```diff
@@ -220,7 +223,28 @@ def extract_netlist_files(netlist_dir,netlist_file):
+
+def start_viewer(working_dir, json_file):
+    viewer_dir = pathlib.Path(__file__).resolve().parent.parent / 'Viewer'
+    shutil.copytree(viewer_dir, working_dir/'Viewer')
```
Need to check if this Viewer directory exists or not. I get errors when I run it twice.
Might be better just to create the directory and soft link in index.html and js/
Then create a new INPUT directory. For me I have a very large INPUT directory and I don't want that copied.

stevenmburns commented on align/main.py:
```diff
@@ -220,7 +223,28 @@ def extract_netlist_files(netlist_dir,netlist_file):
+    stderr = sys.stderr
+    Handler = functools.partial(http.server.SimpleHTTPRequestHandler, directory=str(working_dir/'Viewer'))
+    with socketserver.TCPServer(('localhost', 0), Handler) as httpd:
+        logger.info(f'Please visit localhost:{httpd.server_address[1]} in your browser')
```
You should include the `?design=telescopic_ota_0` part of the url as well in this comment string.
Then you just need to cut and paste.

stevenmburns reviewed (changes requested):
Please make these changes. Good idea. Will be great to have this work.

soneryaldiz commented:
@stevenmburns , addressed your review in the last commit. There is still a problem. When I run and hit Ctrl+C, flow continues nicely. If I visit the url in a browser, Ctrl+C is not throwing an exception as expected. I am trying to resolve this.

stevenmburns reviewed (approved):
Looks great.

stevenmburns commented:
> @stevenmburns , addressed your review in the last commit. There is still a problem. When I run and hit Ctrl+C, flow continues nicely. If I visit the url in a browser, Ctrl+C is not throwing an exception as expected. I am trying to resolve this.

Don't know about this. I guess we can kill it as a workaround.

soneryaldiz commented:
> > @stevenmburns , addressed your review in the last commit. There is still a problem. When I run and hit Ctrl+C, flow continues nicely. If I visit the url in a browser, Ctrl+C is not throwing an exception as expected. I am trying to resolve this.
> 
> Don't know about this. I guess we can kill it as a workaround.

Ctrl+C does not work in WSL+Chrome but works in native Linux+Firefox. I will proceed with merge.
